File: log_redactor/IPv4Generator.py
import ipaddress

class IPv4Generator:
    def __init__(self):
        # Define the base of the 240.0.0.0/4 range
        self.base_ipv4 = ipaddress.IPv4Address("240.0.0.0")
        # Calculate the total number of addresses in the 240.0.0.0/4 range (2^(32 - 4))
        self.max_addresses = 2 ** (32 - 4)
        self.ipv4_counter = 0

    def generate_unique_ipv4(self) -> str:
        """Generate a unique IPv4 address within the 240.0.0.0/4 range."""
        if self.ipv4_counter >= self.max_addresses:
            raise ValueError("No more IPv4 addresses available in the 240.0.0.0/4 range.")

        # Generate the next IPv4 address
        mapped_ipv4 = self.base_ipv4 + self.ipv4_counter
        self.ipv4_counter += 1
        return str(mapped_ipv4)

# Example usage
if __name__ == "__main__":
    ipv4_gen = IPv4Generator()
    for _ in range(10):  # Generate 10 IPv4 addresses for demonstration
        print(ipv4_gen.generate_unique_ipv4())

File: src/ipv4_generator.rs
use std::net::Ipv4Addr;

pub struct IPv4Generator {
    base_ipv4: Ipv4Addr,
    ipv4_counter: u32,
    max_addresses: u32,
}

impl IPv4Generator {
    pub fn new() -> Self {
        IPv4Generator {
            base_ipv4: Ipv4Addr::new(240, 0, 0, 0),
            ipv4_counter: 0,
            max_addresses: 2u32.pow(28), // 240.0.0.0/4 range
        }
    }

    pub fn generate_unique_ipv4(&mut self) -> Result<String, String> {
        if self.ipv4_counter >= self.max_addresses {
            return Err("No more IPv4 addresses available in the 240.0.0.0/4 range.".to_string());
        }
        let mapped_ipv4 = self.base_ipv4.octets();
        let new_ip = Ipv4Addr::new(
            mapped_ipv4[0],
            mapped_ipv4[1],
            mapped_ipv4[2],
            mapped_ipv4[3] + self.ipv4_counter as u8,
        );
        self.ipv4_counter += 1;
        Ok(new_ip.to_string())
    }
}
